tests/testthat/test-subscript-loc.R

```diff
@@ -401,6 +401,91 @@ test_that("can disallow missing values", {
   })
 })
 
+test_that("can alter logical missing value handling (#1595)", {
+  x <- c(TRUE, NA, FALSE, NA)
```

Comment:
Probably best to use a named vector in the logical and integer cases too, as you did in the character case.

Reply:
That revealed a few more places where this comes up
https://github.com/r-lib/vctrs/pull/1633/commits/d3ca2ad8e2eae8b2d53cabaf3b46b87c3db8ae49

I've prepped for the merge of r-lib/rlang#1471, ill leave this open until we can rebase on that to fix these tests
https://github.com/r-lib/vctrs/pull/1633/commits/c1c6f2c3d7b60201aadb57976322e8ad44bc7975